app/controllers/dorms_controller.rb

```diff
@@ -212,6 +214,18 @@ def destroy
   end
 
   private
+    def current_draw_period
```

Comment:
These seem like good changes, but if you are not aware of making please at least try to figure out where they came from.  And I would like them to be tested a bit before we merge this.